### DISCUSSION TITLE
Isn't using meta in useHead responsive?

### DISCUSSION THREAD

**Discussion by xhc-code** [nuxt/nuxt#28907] [Questions]
https://nuxt.com/docs/getting-started/seo-meta
![image](https://github.com/user-attachments/assets/3a216e05-b68b-4d48-a918-285b0188549c)

```ts
<script setup lang="ts">
const route = useRoute()

useHead({
  meta: [{ property: 'og:title', content: `App Name - ${route.meta.title}` }]
})
</script>
```
Meta has been set in the layout page, which does not dynamically change and is only loaded once upon the first browser request. When navigating to other pages through VueRouter, the route changes, but the meta value in useHead does not dynamically change. Is this functional?
Since it won't change, why not just write it on a specific page? Instead of a single layout page (as a layout page will serve as a shared layout structure among multiple pages).

**Comment by joerosenthal** [comment]
My understanding is that you need to use a computed getter to have reactivity on useHead for your example.

In this case I’d probably do useSeoMeta since you’re only modifying the meta data, and it’s a bit cleaner:
```vue
useSeoMeta({
  ogTitle: () => `App Name - ${route.meta.title}`
})
```

**Reply by xhc-code** [reply]
I don't know if the properties in useHead support responsiveness. You are using the useSeoMeta function, which belongs to another way, right.

**Reply by joerosenthal** [reply]
useHead is responsive and can be reactive with a computed getter on an entire reactive object going into the function or you can use the computed getter on each reactive property. useSeoMeta needs to have a computed getter on each reactive property to have reactivity. I do both on my production Nuxt 3 site. 

Why not just try the example I gave you?

**Reply by xhc-code** [reply]
Okay, thank you, big shot.

**Reply by joerosenthal** [reply]
I misspoke in describing how to use useSeoMeta and useHead with reactive data, and I updated my comment accordingly. This is not something I've seen in documentation, but I've tested this extensively.
